catalog/ui/search-ui/standard/src/main/java/org/codice/ddf/ui/searchui/standard/properties/ConfigurationStore.java

```diff
@@ -266,21 +262,71 @@ public void setTerrainProvider(String terrainProvider) {
     }
 
     private void setProxiesForImagery(List<String> imageryProviders) {
+        List<Map<String, Object>> newImageryProviders = imageryProviders.stream()
+                .map(this::getProviderConfigMap)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+
+        List<Map<String, Object>> imageryProvidersToStop = new ArrayList<>();
+        List<Map<String, Object>> imageryProvidersToStart = new ArrayList<>();
+
+        findDifferences(proxiedImageryProviders, newImageryProviders, imageryProvidersToStart);
+        findDifferences(newImageryProviders, proxiedImageryProviders, imageryProvidersToStop);
+
+        List<String> proxiesToStop = imageryProvidersToStop.stream()
+                .map(provider -> urlToProxyMap.get(provider))
+                .collect(Collectors.toList());
+
+        stopImageryEndpoints(proxiesToStop);
+        startImageryEndpoints(imageryProvidersToStart);
+        proxiedImageryProviders = newImageryProviders;
+    }
+
+    private void findDifferences(List<Map<String, Object>> innerList,
```

Comment:
Options to simplify:
1. This one is a little less efficient but really clear
```java
differences.addAll(outerList);
differences.removeIf(innerList::contains);
```
2. This one is more efficient
```java
differences.addAll(outerList.stream()
    .filter(x -> !innerList.contains(x))
    .collect(Collectors.toList()));
```

(I would double check to be sure I didn't invert the inner and outer on those two examples.)